src/dex/nerve/nerve-pool.ts

```diff
@@ -0,0 +1,433 @@
+    state.adminFee = bigIntify(event.args.adminFee);
+    state.defaultWithdrawFee = bigIntify(event.args.withdrawFee);
+
+    return state;
+  }
+
+  handleNewAdminFee(event: any, state: PoolState) {
+    if (!state.isValid) return null;
+
+    state.adminFee = bigIntify(event.args.newAdminFee);
+    return state;
+  }
+
+  handleNewSwapFee(event: any, state: PoolState) {
+    if (!state.isValid) return null;
+
+    state.swapFee = bigIntify(event.args.newSwapFee);
+    return state;
+  }
+
+  handleNewDepositFee(event: any, state: PoolState) {
+    if (!state.isValid) return null;
+
+    state.defaultDepositFee = bigIntify(event.args.newDepositFee);
+    return state;
+  }
+
+  handleNewWithdrawFee(event: any, state: PoolState) {
+    if (!state.isValid) return null;
+
+    state.defaultWithdrawFee = bigIntify(event.args.newWithdrawFee);
+    return state;
+  }
+
+  handleRampA(event: any, state: PoolState) {
+    if (!state.isValid) return null;
+
+    state.initialA = bigIntify(event.args.oldA);
+    state.futureA = bigIntify(event.args.newA);
+    state.initialATime = bigIntify(event.args.initialTime);
+    state.futureATime = bigIntify(event.args.futureTime);
+    return state;
+  }
+
+  handleStopRampA(event: any, state: PoolState) {
+    if (!state.isValid) return null;
+
+    // To support IronV2 variable names I use "or" expression
+    const finalA = bigIntify(event.args.currentA || event.args.A);
+    const finalTime = bigIntify(event.args.time || event.args.timestamp);
+
+    state.initialA = finalA;
+    state.futureA = finalA;
+    state.initialATime = finalTime;
+    state.futureATime = finalTime;
+    return state;
+  }
+
+  handleTokenSwap(
+    event: any,
+    state: PoolState,
+    _2: Log,
+    blockHeader: BlockHeader,
+  ) {
+    if (!state.isValid) return null;
+    const blockTimestamp = bigIntify(blockHeader.timestamp);
+
+    const transferredDx = bigIntify(event.args.tokensSold);
+    const dyEvent = bigIntify(event.args.tokensBought);
+    const tokenIndexFrom = event.args.soldId.toNumber();
+    const tokenIndexTo = event.args.boughtId.toNumber();
+
+    try {
+      const swap = this.math.calculateSwap(
+        state,
+        tokenIndexFrom,
+        tokenIndexTo,
+        transferredDx,
+        blockTimestamp,
+      );
+      const { dy, dyFee } = swap;
+
+      const dyAdminFee =
+        (dyFee * state.adminFee) /
+        this.math.FEE_DENOMINATOR /
+        state.tokenPrecisionMultipliers[tokenIndexFrom];
+
+      state.balances[tokenIndexFrom] += transferredDx;
+      state.balances[tokenIndexTo] -= dy + dyAdminFee;
+
+      if (dyEvent !== dy) {
+        this.logger.error(
+          `For ${this.parentName}_${this.poolConfig.name} _calculateSwap value ${dy} is not equal to ${dyEvent} event value`,
+        );
+        state.isValid = false;
+        return null;
+      }
+
+      return state;
+    } catch (e) {
+      state.isValid = false;
+      return null;
```

Comment:
You can't return `null` if you need to change the state (by setting `isValid` to `false` for example) because then the change won't be processed.  Return `state` instead.  (Happens in a lot of places.)